### PR TITLE
Add JSdocs comments

### PR DESCRIPTION
**Pull request by dpmittal**
### Problem

Documentation is a very important part in Open source project which is not present in the document and it makes it difficult for new developers to collaborate as well as existing developers to remember what they wrote and why they wrote it.

### Solution

Adding JSdocs comments to methods and their parameters and return values make the code understandable.

### Areas of Impact

JSdocs comments will not affect any part of the codebase

### CODE REVIEW

File: src/client/components/pages/about.js [hunk 1]
```diff
@@ -110,7 +115,7 @@ function AboutPage() {
 			<p>
 				The core components of BookBrainz are the site, the
 				webservice and the database. All of these components are
-				running on MetaBrainz's Sakura server, which also runs&nbsp;
+				running on MetaBrainz&#39;s Sakura server, which also runs&nbsp;
```
Comment (LordSputnik) [inline src/client/components/pages/about.js]: Good change, but it would be better to use &rsquo; - more typographically correct and easier to remember the HTML character code.

File: src/client/components/pages/about.js [hunk 2]
```diff
@@ -15,11 +15,16 @@
  * with this program; if not, write to the Free Software Foundation, Inc.,
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/* eslint valid-jsdoc: ["error", { "requireReturn": false }] */
```
Comment (LordSputnik) [inline src/client/components/pages/about.js]: Let's update the rules globally for this, since we seem to be making exceptions everywhere for it. It seems like the right thing to be doing.

File: src/client/components/pages/about.js [hunk 3]
```diff
@@ -15,11 +15,16 @@
...
 
 const React = require('react');
 
 const PageHeader = require('react-bootstrap').PageHeader;
 
+/**
+* Renders the document and display the 'About' Page.
```
Comment (LordSputnik) [inline src/client/components/pages/about.js]: "displays" rather than "display" and also lowercase "page".

File: src/client/components/pages/entity-revisions.js [hunk 4]
```diff
@@ -24,14 +26,28 @@ const Col = bootstrap.Col;
 const ListGroup = bootstrap.ListGroup;
 const ListGroupItem = bootstrap.ListGroupItem;
 
+/**
+* This is an extended class of the React Component to
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: "This class is derived from the React Component base class and renders..." would be better, I think.

File: src/client/components/pages/entity-revisions.js [hunk 5]
```diff
@@ -24,14 +26,28 @@ const Col = bootstrap.Col;
...
+* render the 'Entity Revisions' page.
+*/
 class EntityRevisions extends React.Component {
 
+/**
+* Binds the class methods to their respective data.
+* @constructor
+* @param {object} props - Object of properties passed in the constructor
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: "the properties passed to the component" would be better here

File: src/client/components/pages/entity-revisions.js [hunk 6]
```diff
@@ -24,14 +26,28 @@ const Col = bootstrap.Col;
...
+*/
 	constructor(props) {
 		super(props);
 		this.renderHeader = this.renderHeader.bind(this);
 		this.renderRevision = this.renderRevision.bind(this);
 	}
 
+/**
+* Renders the Header of the page consisting of Revision ID and
+* entity information in the form of a table.
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: This isn't quite correct - there is no table involved - only the entity name and disambiguation comment are rendered here.

File: src/client/components/pages/entity-revisions.js [hunk 7]
```diff
@@ -54,6 +70,12 @@ class EntityRevisions extends React.Component {
 		);
 	}
 
+/**
+* Renders the data related to Revision such as 'author' and 'date'.
+* It also displays 'Revision Notes' column.
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: It actually displays the first revision note, which should be a summary of the changes made in the revision

File: src/client/components/pages/entity-revisions.js [hunk 8]
```diff
@@ -54,6 +70,12 @@ class EntityRevisions extends React.Component {
...
+* @param {object} revision object is passed for it's data to be accessed
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: "the revision to be represented by the rendered component" would be better here

File: src/client/components/pages/entity-revisions.js [hunk 9]
```diff
@@ -83,6 +105,11 @@ class EntityRevisions extends React.Component {
 		);
 	}
 
+/**
+* Renders the 'Revision History' column of the EntityRevisions page
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: I'd have something like this here: "Renders the EntityRevisions page, which is a list of all the revisions made to an entity, along with information about the author and the first revision note for each revision"

**Comment by dpmittal** [comment]
@LordSputnik I have made the suggested changes :D

File: src/client/components/pages/about.js (hunk 1 above)
Comment (LordSputnik) [inline src/client/components/pages/about.js]: Leaving this as it was is fine, but it seems that GitHub changed what I meant to say - that using `&rsquo;` is better (rather than `'` or `&#39;`).

File: src/client/components/pages/contribute.js [hunk 10]
```diff
@@ -15,11 +15,16 @@
...
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/* eslint valid-jsdoc: "error" */
 
 const React = require('react');
 
 const PageHeader = require('react-bootstrap').PageHeader;
 
+/**
+* Renders the the document and display the 'Contribute' Page.
```
Comment (LordSputnik) [inline src/client/components/pages/contribute.js]: This should also be "displays"

File: src/client/components/pages/develop.js [hunk 11]
```diff
@@ -15,11 +15,16 @@
...
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/* eslint valid-jsdoc: "error" */
 
 const React = require('react');
 
 const PageHeader = require('react-bootstrap').PageHeader;
 
+/**
+* Renders the document and display the 'Develop' Page.
```
Comment (LordSputnik) [inline src/client/components/pages/develop.js]: This should also be "displays"

File: src/client/components/pages/contribute.js [hunk 12]
```diff
@@ -15,11 +15,16 @@
...
+* @returns {ReactElement} a HTML document which displays the contribute page
```
Comment (LordSputnik) [inline src/client/components/pages/contribute.js]: "the Contribute page" - you've capitalised elsewhere, so it makes sense to be consistent

File: src/client/components/pages/about.js (hunk 1 above)
Comment (dpmittal) [inline src/client/components/pages/about.js]: I have made the change :D

File: src/client/components/pages/entity-revisions.js [hunk 13]
```diff
@@ -24,14 +26,28 @@ const Col = bootstrap.Col;
 const ListGroup = bootstrap.ListGroup;
 const ListGroupItem = bootstrap.ListGroupItem;
 
+/**
+* The class is derived from React Component base class and
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: "from *the* React Component..." makes more sense here

File: src/client/components/pages/entity-revisions.js [hunk 14]
```diff
@@ -24,14 +26,28 @@ const Col = bootstrap.Col;
...
+* renders the 'Entity Revisions' page.
+*/
 class EntityRevisions extends React.Component {
 
+/**
+* Binds the class methods to their respective data.
+* @constructor
+* @param {object} props - Object of properties passed to the component
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: No need for "Object of" here

File: src/client/components/pages/entity-revisions.js [hunk 15]
```diff
@@ -24,14 +26,28 @@ const Col = bootstrap.Col;
...
 	constructor(props) {
 		super(props);
 		this.renderHeader = this.renderHeader.bind(this);
 		this.renderRevision = this.renderRevision.bind(this);
 	}
 
+/**
+* Renders the Header of the page consisting of Entity name and
+* disambiguation comment.
+* @returns {ReactElement} a HTML document which is a a part of Revision page
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: Repeated "a" in this comment. Also it would make more sense if you wrote "is a part of the Revision page"

File: src/client/components/pages/entity-revisions.js [hunk 16]
```diff
@@ -54,6 +70,14 @@ class EntityRevisions extends React.Component {
 		);
 	}
 
+/**
+* Renders the data related to Revision such as 'author' and 'date'.
+* It also displays 'Revision Notes' column which displays the first revison note
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: No columns involved here - it renders the first revision note for each revision inside a list group item. The function is used in a loop in `render()` to render all of the revisions inside a list group.

File: src/client/components/pages/entity-revisions.js [hunk 17]
```diff
@@ -54,6 +70,14 @@ class EntityRevisions extends React.Component {
...
+* which is a summary of the changes made in the revision.
+* @param {object} revision object is passed for the revison to be represented
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: No need for "object is passed" - that's implied by `revision` being a parameter for the function.

File: src/client/components/pages/entity-revisions.js [hunk 18]
```diff
@@ -83,6 +107,12 @@ class EntityRevisions extends React.Component {
 		);
 	}
 
+/**
+* Renders the EntityRevisions page, which is a list of all the revisions made
+* to an entity, along with information about the author and the
+* first revision note for each revision.
+* @returns {ReactElement} a HTML document which is a part of the Revision page
```
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: "a HTML document which displays the Revision page" - this renders the whole thing, not just a part

File: src/client/components/pages/about.js [hunk 19]
```diff
@@ -15,11 +15,16 @@
  * with this program; if not, write to the Free Software Foundation, Inc.,
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+/* eslint valid-jsdoc: "error" */
```
Comment (LordSputnik) [inline src/client/components/pages/about.js]: What I meant here was it would be better to add the `requiresReturn: false` option to the `.eslintrc` file in the root directory of the repository, to save us from having to add a comment at the top of each file.

File: src/client/components/pages/about.js [hunk 20]
```diff
@@ -110,7 +115,8 @@ function AboutPage() {
 			<p>
 				The core components of BookBrainz are the site, the
 				webservice and the database. All of these components are
-				running on MetaBrainz's Sakura server, which also runs&nbsp;
```
Comment (Freso) [inline src/client/components/pages/about.js]: Why are you changing this line?

File: src/client/components/pages/entity-revisions.js [hunk 21]
```diff
@@ -54,6 +69,14 @@ class EntityRevisions extends React.Component {
 		);
 	}
 
+/**
+* Renders the data related to Revision such as 'author' and 'date'.
+* It also displays the first revison note which is a summary of the changes
+* made in the revision.
+* @param {object} revision for the revison to be represented  by the
```
Comment (Freso) [inline src/client/components/pages/entity-revisions.js]: Two spaces between "represented" and "by".

File: src/client/components/pages/entity-revisions.js [hunk 22]
```diff
@@ -16,6 +16,7 @@
  * with this program; if not, write to the Free Software Foundation, Inc.,
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
+
```
Comment (Freso) [inline src/client/components/pages/entity-revisions.js]: It is usually better to keep code style changes (such as whitespace) in separate commits (or even completely separate PRs), so they can be reviewed outside of "actual" code changes.

File: src/client/components/pages/entity-revisions.js [hunk 23]
```diff
@@ -24,14 +25,28 @@ const Col = bootstrap.Col;
...
+/**
+* The class is derived from the React Component base class and
+* renders the 'Entity Revisions' page.
+*/
 class EntityRevisions extends React.Component {
 
+/**
+* Binds the class methods to their respective data.
+* @constructor
+* @param {object} props - Properties passed to the component
```
Comment (Freso) [inline src/client/components/pages/entity-revisions.js]: This `@param` is formatted differently than the one for `renderRevision`. Is this intentional? Did you purposefully uppercase "Properties"?

File: src/client/components/pages/about.js (hunk 20 above)
Comment (dpmittal) [inline src/client/components/pages/about.js]: It improved the readability of the code. @LordSputnik  has approved these changes you can see his 3rd comment from top for reference :)

File: src/client/components/pages/entity-revisions.js (hunk 23 above)
Comment (dpmittal) [inline src/client/components/pages/entity-revisions.js]: This `@param` is formatted correctly but the `renderRevsion` so I'm changing that now. Properties hasn't been purposely uppercased. Should I lowercase it?

File: src/client/components/pages/entity-revisions.js (hunk 21 above)
Comment (dpmittal) [inline src/client/components/pages/entity-revisions.js]: I'm changing it now :) it's a silly mistake

File: src/client/components/pages/entity-revisions.js (hunk 22 above)
Comment (dpmittal) [inline src/client/components/pages/entity-revisions.js]: It was just one line addition so i thought it to include it in this PR? Should I remove it and send a separate PR?

Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: Shouldn't need to add this line at all - it's just left over from the eslint comment you added earlier.

File: src/client/components/pages/about.js (hunk 20 above)
Comment (LordSputnik) [inline src/client/components/pages/about.js]: I don't mind this change - not part of the changes mentioned in the PR, so generally it would be better to keep it separate (don't worry for now), but it's an improvement nonetheless.

File: src/client/components/pages/entity-revisions.js (hunk 23 above)
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: http://usejsdoc.org/tags-param.html uses uppercase `@param` descriptions, so let's follow that (although it will vary from project to project).

File: src/client/components/pages/entity-revisions.js (hunk 21 above)
Comment (LordSputnik) [inline src/client/components/pages/entity-revisions.js]: I'd also remove the "for" here, and have the description start with "the revision..."

**Comment by dpmittal** [comment]
@LordSputnik Made the changes :)

**Comment by LordSputnik** [comment]
FYI, your docs are now public at https://doclets.io/bookbrainz/bookbrainz-site/master - that'll update automatically whenever any new ones are added!
